tests/docs/test_run_generate_dataset_example.py

```diff
@@ -0,0 +1,92 @@
+import textwrap
+import unittest
+
+import pytest
+
+import ehrql.loaders
+
+from . import test_complete_examples
+
+
+def test_run_generate_dataset_example(tmp_path):
+    example = test_complete_examples.DatasetDefinitionExample(
+        path="test",
+        fence_number=1,
+        source=textwrap.dedent(
+            """\
+            from ehrql import create_dataset
+            from ehrql.tables.beta.tpp import patients
+
+            dataset = create_dataset()
+            dataset.define_population(patients.exists_for_patient())
+            """
+        ),
+    )
+    test_complete_examples.test_ehrql_generate_dataset_example(tmp_path, example)
+
+
+def test_run_generate_dataset_example_failing(tmp_path):
+    example = test_complete_examples.DatasetDefinitionExample(
+        path="test",
+        fence_number=1,
+        source=textwrap.dedent(
+            """\
+            from ehrql import create_dataset
+
+            dataset = create_dataset()
+            dataset.define_population(not_a_function())
+            """
+        ),
+    )
+    with pytest.raises(ehrql.loaders.DefinitionError):
+        test_complete_examples.test_ehrql_generate_dataset_example(tmp_path, example)
+
+
+def test_run_generate_dataset_example_failing_codelist_from_csv_call(tmp_path):
+    example = test_complete_examples.DatasetDefinitionExample(
+        path="test",
+        fence_number=1,
+        source=textwrap.dedent(
+            """\
+            from ehrql import codelist_from_csv, create_dataset
+            from ehrql.tables.beta.tpp import patients
+
+            codes = codelist_from_csv()
+
+            dataset = create_dataset()
+            dataset.define_population(patients.exists_for_patient())
+            """
+        ),
+    )
+
+    with pytest.raises(ehrql.loaders.DefinitionError) as e:
+        test_complete_examples.test_ehrql_generate_dataset_example(tmp_path, example)
+        # This should only fail if the test fails, so skip coverage.
+        assert (e.__context__) == TypeError(  # pragma: no cover
+            "codelist_from_csv() given incorrect arguments"
+        )
+
+
+def test_run_generate_dataset_example_gives_unreadable_csv(tmp_path):
```

Comment:
thought: There are potentially more tests we could add here; for example, checking that ` ```{.ehrql} ` works, but as we're not using that feature yet — maybe we would if wanting to add extra inline data — I haven't.